src/lava/magma/core/model/py/type.py

```diff
@@ -12,4 +18,104 @@ class LavaPyType:
         type, ty.Type[PyInPort], ty.Type[PyOutPort], ty.Type[PyRefPort]
     ]
     d_type: type
-    precision: int = None  # If None, infinite precision is assumed
+    precision: str = None  # If None, infinite precision is assumed.
+    num_bits_exp: int = None  # If None, fixed exponent in fixed-pt model.
+    exp_var: str = None  # Name of Var in which exponent is stored, if needed.
+    domain: np.ndarray = None  # If None, no a-priori knowledge of Var domain.
+    constant: bool = False  # If True, indicates that Var is constant.
+    meta_parameter: bool = False  # If True, indicates that Var is config var
+    # Indicate to which scale domain Var belongs, i.e., with which
+    # Vars scaling needs to be consistent. Scale domains are identified by
+    # integers: '0' indicates the global scale domain shared between all
+    # Processes. Other integers are reserved for scale domains local to one
+    # Process.
+    # By default all Vars are assumed to be in same scale domain.
+    scale_domain: int = 0
+
+    @staticmethod
+    def _validate_precision(precision):
+        """Validates the format of the precision passed to the LavaPyType.
+
+        Parameters
+        ----------
+        precision : string
+            Specific precision in format '{u, s}:x:y', x, y ints
+
+        Raises
+        ------
+        Warning
+            If precision is None, might cause error in Float2Fixed Point
+            Conversion
+        TypeError
+            If precision is not of type str or None
+        ValueError
+            If precision passed in wrong format
+        """
+
+        if precision is None:
+            warnings.warn("'precision' is None: This might cause an error in"
+                          + " the Float2FixedPoint conversion")
+        elif type(precision) is not str:
+            raise TypeError("'precision' must be of type str or None")
```

Comment:
When raising an error it is best to include the actual value of the variable for debugging. Thus, I would here add the precision variable. Also below in the ValueErrors. Something of the form

```
raise TypeError(f"'precision' must be of type str or None, but'"
    f"{is of type type(precision)}'")
```

Reply:
Addressed in 96d55e369307c301015c073d3ccec2049e89b006.